Formula/onedrive.rb

```diff
@@ -0,0 +1,33 @@
+class Onedrive < Formula
+  desc "Folder synchronization with OneDrive"
+  homepage "https://github.com/abraunegg/onedrive"
+  url "https://github.com/abraunegg/onedrive/archive/v2.4.12.tar.gz"
+  sha256 "4f6aa46fc28e859b63c0a2c190f66c3286ec0bde3f54f77af0ddea6b62bba37a"
+  license "GPL-3.0-only"
+
+  depends_on "dmd" => :build
+  depends_on "pkg-config" => :build
+  depends_on "curl"
+  depends_on :linux
+  depends_on "sqlite"
+
+  on_linux do
+    patch do
+      url "https://github.com/abraunegg/onedrive/commit/761cf3eb878fd370576329127055eec06b975672.patch?full_index=1"
+      sha256 "9a892e5e6bed7695754d49f9561d59f22180db7340493ec7f167bee767b8c28e"
+    end
+  end
+
+  def install
+    ENV["DC"] = "dmd"
+    system "./configure", "--prefix=#{prefix}"
+    system "make", "install"
+    bash_completion.install "contrib/completions/complete.bash" => "onedrive"
+    zsh_completion.install "contrib/completions/complete.zsh" => "_onedrive"
+    fish_completion.install "contrib/completions/complete.fish" => "onedrive.fish"
+  end
+
+  test do
+    system "#{bin}/onedrive", "--version"
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
I can make better tests, but it could take some time. Any help would be appreciated.

Reply:
Added test in jnooree/homebrew-cask@7dfc49a.